### PR TITLE
Replacing the stats stub struct in our class struct with playerstat's actual stat and effects hash 

### PR DESCRIPTION
**Pull request by simrenk**
Having been tasked with implementing any functionality rpg-NPC needed in order to be able to set an NPC to a specific class (#670), I added playerstats actual stats and effects structs into our struct class to replace our hardcoded stats struct. I adjusted our class_new and class_init functions accordingly.

This not only resolves rpg-NPC build issues but also redundancy issues (noted by rpg-NPC in #699) because @anshikab @avianurag @carlosazpurua1004 and I all decided it would make the most sense for the class struct to contain the stat struct.

In order to ensure that previously merged files using the old class_new and class_init function still compiled + tests still worked, I updated other teams' tests as well.

### CODE REVIEW

File: include/playerclass/class.h [hunk 1]
```diff
@@ -25,8 +26,8 @@
  *  - NULL on error
  */
 class_t* class_new(char* name, char* shortdesc, char* longdesc,
-                   obj_t* attr, stats_t* stat, skilltree_t* skilltree,
-                   skill_t* combat, skill_t* noncombat);
+                   obj_t* attr, stats_hash_t* stat, effects_hash_t* effect,
+		   skilltree_t* skilltree, skill_t* combat, skill_t* noncombat);
```
Comment (ethanwtai) [inline include/playerclass/class.h:30]: Try and keep this on the same line as the others

Comment (ethanwtai) [reply include/playerclass/class.h:30]: Also there are tab characters here

File: include/playerclass/class.h [hunk 2]
```diff
@@ -46,8 +48,8 @@ class_t* class_new(char* name, char* shortdesc, char* longdesc,
  *  - EXIT_FAILURE otherwise
  */
 int class_init(class_t* class, char* name, char* shortdesc, char* longdesc,
-               obj_t* attr, stats_t* stat, skilltree_t* skilltree,
-               skill_t* combat, skill_t* noncombat);
+               obj_t* attr, stats_hash_t* stat, effects_hash_t* effect,
+	       skilltree_t* skilltree, skill_t* combat, skill_t* noncombat);
```
Comment (ethanwtai) [inline include/playerclass/class.h:52]: Fix formatting

File: src/playerclass/src/class.c [hunk 3]
```diff
@@ -14,8 +14,8 @@
 
 /* See class.h */
 class_t* class_new(char* name, char* shortdesc, char* longdesc,
-                   obj_t* attr, stats_t* stat, skilltree_t* skilltree,
-                   skill_t* combat, skill_t* noncombat)
+                   obj_t* attr, stats_hash_t* stat, effects_hash_t* effect,
+  		    skilltree_t* skilltree, skill_t* combat, skill_t* noncombat)
```
Comment (ethanwtai) [inline src/playerclass/src/class.c:18]: Fix formatting (tab characters and same line

File: src/playerclass/src/class.c [hunk 4]
```diff
@@ -39,8 +39,8 @@ class_t* class_new(char* name, char* shortdesc, char* longdesc,
 
 /* See class.h */
 int class_init(class_t* class, char* name, char* shortdesc, char* longdesc,
-                   obj_t* attr, stats_t* stat, skilltree_t* skilltree,
-                   skill_t* combat, skill_t* noncombat)
+	       obj_t* attr, stats_hash_t* stat, effects_hash_t* effect,
+	        skilltree_t* skilltree, skill_t* combat, skill_t* noncombat)
```
Comment (ethanwtai) [inline src/playerclass/src/class.c:43]: Fix formatting

File: tests/battle/test_battle_flow.c [hunk 5]
```diff
@@ -50,7 +50,7 @@ Test(battle_flow, set_one_enemy)
 {
     class_t* test_class = class_new("Bard", "Music boi",
                                     "Charismatic, always has a joke or song ready",
-                                     NULL, NULL, NULL, NULL, NULL);
+				    NULL, NULL, NULL, NULL, NULL, NULL);
```
Comment (ethanwtai) [inline tests/battle/test_battle_flow.c:53]: Fix formatting

**Comment by dwahme** [comment]
Issue Score: ✔️+

Comments:
Great job! Here are a few things you may want to improve in future issues:

When an issue is done, you should include a final comment summarizing what was done, and why the issue is being closed.

**Pull request merged**